### PR TITLE
Feature/add dialog edit functionality

### PR DESCRIPTION
**Pull request by sebavuye**

### CODE REVIEW

File: src/components/Dialog/index.tsx [hunk 1]
```diff
@@ -8,52 +9,83 @@ import {
...
+	DIALOG_TITLE_EDIT,
 	VALIDATION_EMAIL_REQUIRED_TEXT,
 	VALIDATION_EMAIL_TEXT,
 	VALIDATION_NAME_REQUIRED_TEXT
 } from "../../constants";
-import Button from "../Button";
-import TextField from "./TextField";
-import { usePlayerData } from "../../utils";
 
+// ---
```
Comment (mahammedzkhan) [inline src/components/Dialog/index.tsx:28]: Empty comment?

File: src/components/Dialog/index.tsx [hunk 2]
```diff
@@ -8,52 +9,83 @@ import {
...
+	const { dialogState, setDialogState } = useDialogState();
+
+	const addPlayerData = (newValues: DialogFormValues): DialogFormValues[] => {
+		const newPlayerData = { ...newValues, id: newValues.id === "" ? nanoid() : newValues.id };
+		const currentPlayerData = [...playerData];
+		return [...currentPlayerData, newPlayerData];
+	};
+
+	const updatePlayerData = (updatedValues: DialogFormValues): DialogFormValues[] => {
+		const matchedPlayerDataIndex = playerData.findIndex(player => player.id === updatedValues.id);
```
Comment (mahammedzkhan) [inline src/components/Dialog/index.tsx:51]: You can also use map and change the values directly.

File: src/components/Dialog/index.tsx [hunk 3]
```diff
@@ -8,52 +9,83 @@ import {
...
+		const currentPlayerData = [...playerData];
+		currentPlayerData[matchedPlayerDataIndex] = updatedValues;
+		return [...currentPlayerData];
+	};
 
-	const updatePlayerData = (newValues: DialogFormValues): DialogFormValues[] => {
+	const deletePlayerData = (): DialogFormValues[] => {
+		const matchedPlayerDataIndex = playerData.findIndex(player => player.id === dialogState.id);
```
Comment (mahammedzkhan) [inline src/components/Dialog/index.tsx:58]: Same here, you can use map to change the values directly.

File: src/utils/test-utils.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+		<PlayerDataProvider>
+			<DialogContextProvider>{children}</DialogContextProvider>
+		</PlayerDataProvider>
+	);
+};
+
+const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, "wrapper">): RenderResult =>
+	render(ui, { wrapper: AppWithProviders, ...options });
+
+export * from "@testing-library/react";
```
Comment (mahammedzkhan) [inline src/utils/test-utils.tsx:17]: Why would you do this?

Comment (sebavuye) [reply src/utils/test-utils.tsx:17]: To omit providing all context providers in every test.

[](https://testing-library.com/docs/react-testing-library/setup/#custom-render)

Comment (sebavuye) [reply src/utils/test-utils.tsx:17]: https://testing-library.com/docs/react-testing-library/setup/#custom-render

**Pull request merged**